### PR TITLE
Replace app-drawer-overlay-changed event with app-drawer-transitioned

### PR DESCRIPTION
**Pull request by keanulee**
Should cover the use cases in #124

### CODE REVIEW

**Comment by keanulee** [comment]
@ktiedt Will something like this work for you?

**Comment by ktiedt** [comment]
Just tested, and this does get us to a usable state. In actually the on-opened-changed pointer was all we needed in the end (decided to not try to match the animation effects since we don't use them anywhere else yet).

I would suggest adding an object with the state to the transition event if it is to be the only transition event though. That way a user does not have to dig through e.currentTarget to access things that should be relevant to the event itself.

Thanks!

**Comment by keanulee** [comment]
@ktiedt What details do you think would be helpful? I didn't want to expose the private `_DRAWER_STATE` object, but perhaps including the current state of opened and persistent in e.detail (even though they would be accessible through e.target)?

**Comment by ktiedt** [comment]
I generally expect custom-events to contain details relevant to the event, in this case the event w/o details does not answer the question of 'transitioned to what?' - If it were me (as you can see in my PR) I would add the state ie: `{opened: this.opened}` but if you are going with minimalist (which is technically true to what normal HTML events do) approach, making the user dig through the element for this would be on par. In the end either way is usable.

**Comment by keanulee** [comment]
Good point. I think we'll stick with the minimalist approach and leave out the detail. Thanks.

File: app-drawer/app-drawer.html [hunk 1]
```diff
@@ -524,25 +529,47 @@
...
+            this._DRAWER_STATE.OPENED_PERSISTENT : this._DRAWER_STATE.OPENED;
         } else {
-          document.removeEventListener('keydown', this._boundKeydownHandler);
-          document.body.style.overflow = '';
+          this._drawerState = this._DRAWER_STATE.CLOSED;
         }
 
-        this.fire('app-drawer-overlay-changed', { overlay: this.opened && !this.persistent });
+        if (oldState !== this._drawerState) {
+          if (this.opened && !this.persistent) {
```
Comment (frankiefu) [inline app-drawer/app-drawer.html]: Is this the same as `if (this._drawerState === this._DRAWER_STATE.OPENED) {`?

Comment (keanulee) [inline app-drawer/app-drawer.html]: I guess it is - will change.

**Comment by frankiefu** [comment]
LGTM

**Comment by ktiedt** [comment]
Thanks!

**Pull request merged**